app/subcommands/deploy_subcommand.py

```diff
@@ -201,7 +201,9 @@ def _validate_successful_deployment(self, master_service_url, slaves_to_validate
         network = Network()
 
         def all_slaves_registered():
-            return len(self._registered_slave_hostnames(slave_api_url, network)) == len(slaves_to_validate)
+            registered_slave_uids = set(map(Network.rsa_key, self._registered_slave_hostnames(slave_api_url, network)))
```

Comment:
Using public rsa key to identify a host is not reliable since there's nothing preventing two different hosts to have the same rsa key pair. I am trying to use ip or MAC address here. But it's not relevant to this change. I will make another PR to replace `Network.rsa_key`.